pkg/controllers/jobset_controller.go

```diff
@@ -552,29 +553,33 @@ func (r *JobSetReconciler) createHeadlessSvcIfNecessary(ctx context.Context, js
 	var headlessSvc corev1.Service
 	subdomain := GetSubdomain(js)
 	if err := r.Get(ctx, types.NamespacedName{Name: subdomain, Namespace: js.Namespace}, &headlessSvc); err != nil {
-		headlessSvc := corev1.Service{
-			ObjectMeta: metav1.ObjectMeta{
-				Name:      subdomain,
-				Namespace: js.Namespace,
-			},
-			Spec: corev1.ServiceSpec{
-				ClusterIP: "None",
-				Selector: map[string]string{
-					jobset.JobSetNameKey: js.Name,
+		if k8serrors.IsNotFound(err) {
```

Comment:
can you flip this just so we reduce indentation:

```
if !k8serrors.IsNotFound(err) {
  return err
}
// proceed to create headless service
```

Reply:
done